### ISSUE TITLE
EVSE Support

### ISSUE THREAD

**Issue by magico13** [magico13/ha-emporia-vue#60]
I just got my Emporia EVSE/EV Charger (whatever you prefer to call it) installed and can now add support to Home Assistant to control it. Without changes it gets pulled in as a new energy sensor no problem but I've already verified that I can do the following from the API (already added to pyemvue):

1. Enable/disable charging
2. Adjust the desired charging current
3. Adjust the maximum current of the charger

Only 1 and 2 are really useful actionable items for Home Assistant, plus a handful of attributes that would be handy to return but not necessarily make sensors out of (status message, fault message, etc). I know I definitely can enable it to appear as a switch so you can enable/disable it from an automation but I think in order to control the charging speed I'll have to create a service that can be called with an integer number of amps as a parameter. I haven't done that so it might come in a V2, but I can imagine that would be very useful to adapt the charge rate to keep your total power draw at a specific level, to match your solar production for instance.

**Comment by debsahu** [comment]
Love to see the switch component to turn on and of EVSE charging. As per setting the charge current, home assistant has input_number that can be used to get range of acceptable values for charge current.

**Comment by dalklein** [comment]
Thanks for supporting and figuring this out!  I've been wanting to have an L2 EVSE that integrates with my solar and battery controls, and seeing this (along with $500 utility rebate), I'll go ahead and get the Emporia EVSE and Vue system too.   Thanks!

**Comment by magico13** [comment]
With Home Assistant you can choose a variety of energy monitors and EVSEs, even from different brands, so don't feel like you have to get these ones because they're the same brand. If you're actively trying to match excess solar to charge rate you might have a better experience with systems that provide local control.

**Comment by dalklein** [comment]
I agree magico, there are various choices.  I have relatively complicated automation system using a RPI with IOTstack, talking to the inverters, bms's, solar charge controllers, and commanding various wifi plugs & relay boards to match loads to supply and charge different batteries, etc.  I've been using just an L1 EVSE on a smartplug which works fine at the beginning and end of days, or cloudy days,  but I've wanted an L2 that I can control remotely from the RPi with variable charging.  

The Emporia monitor and the EVSE seem to be pretty reasonable or pretty much the lowest cost hardware.  OpenEVSE could be better being totally local, but it's not necessarily less expensive either.  My utility rebate also requires it to be on the EnergyStar list.
 I'm only using serial data reported from the inverters over rs485 which updates only every 10 sec or so, and my controls work even more slowly with a couple minutes of time hysteresis.  So if the Emporia is reporting up to the cloud and back again, that will be ok.      

I will be probably using your PyEmVue only, I don't use HA.   (but this seemed like the EVSE discussion)  If it gets figured out to do with local data only, that would be great too, one day.

Thanks for sharing your work and code here!

**Comment by wirecuttr** [comment]
Hi, I have the Emporia EVSE, but I don't see any way to change the charging current. I just see entities for:
sensor.ev_charger_123_1min
sensor.ev_charger_123_1d
sensor.ev_charger_123_1mon
switch.ev_charger

Is there something I need to do to expose the other EV charger parameters?

**Comment by magico13** [comment]
@wirecuttr changing the current is done via a service call, not via an entity (yet). The yaml for that might look like this, but you can also change it via the Developer Tools UI or use it in an automation from the UI.

```yaml
service: emporia_vue.set_charger_current
data:
  current: 8
target:
  entity_id: switch.ev
```

**Comment by wirecuttr** [comment]
Ah. Thanks. I'm a bit new to all of this.
In this post there is mention of attributes of the charger: https://github.com/magico13/ha-emporia-vue/issues/223#issue-1847268809

Are those also accessed via service call?

**Comment by wirecuttr** [comment]
Thanks for the clarification!

**Comment by magico13** [comment]
@starthal the docs definitely need some updates, but you are correct that EVSE support is mostly complete. There's some stuff that's only exposed via attributes that would be nice to expose as full sensors, but you can turn the EVSE on/off as a standard switch, and can set the charging current via a service call.

**Comment by starthal** [comment]
Thanks!

I will probably be trying this in the next few weeks (I have the "Classic" EVSE but no Vue); I'll make a doc PR if possible.

Is README.md the right spot? Does HACS or something else use the info.md?
